packages/x-tree-view/src/TreeItem/TreeItem.tsx

```diff
@@ -404,6 +404,10 @@ TreeItem.propTypes = {
    * Use the `onItemFocus` callback on the tree if you need to monitor a item's focus.
    */
   onFocus: unsupportedProp,
+  /**
+   * Callback fired when a key of the keyboard is pressed on the item.
```

Comment:
Maybe we can rephrase it something like:
```suggestion
   * Callback fired when a key is pressed on the keyboard and the tree is in focus.
```
🤔